src/bwamem.cpp

```diff
@@ -2127,39 +2115,40 @@ void mem_chain2aln_across_reads_V2(const mem_opt_t *opt, const bntseq_t *bns,
 					sp.seqid = c->seqid;
 					sp.regid = av->n - 1;
 						
-					// assert(numPairsLeft < BATCH_SIZE * SEEDS_PER_READ);
-					if (numPairsLeft >= *wsize) {
-						fprintf(stderr, "[LOG][%0.4d] Re-allocating seqPairArrays Left\n", tid);
-						*wsize += 1000;
-						seqPairArrayAux = (SeqPair *) realloc(seqPairArrayAux,
-															  (*wsize + MAX_LINE_LEN)
-															  * sizeof(SeqPair));
-						mmc->seqPairArrayAux[tid] = seqPairArrayAux;
-						seqPairArrayLeft128 = (SeqPair *) realloc(seqPairArrayLeft128,
-																  (*wsize + MAX_LINE_LEN)
-																  * sizeof(SeqPair));
-						mmc->seqPairArrayLeft128[tid] = seqPairArrayLeft128;
-						seqPairArrayRight128 = (SeqPair *) realloc(seqPairArrayRight128,
-																   (*wsize + MAX_LINE_LEN)
-																   * sizeof(SeqPair));
-						mmc->seqPairArrayRight128[tid] = seqPairArrayRight128;
-					}
-
-					
 					sp.idq = leftQerOffset;
 					sp.idr = leftRefOffset;
-					uint8_t *qs = seqBufLeftQer + sp.idq;
-					uint8_t *rs = seqBufLeftRef + sp.idr;
 					
 					leftQerOffset += s->qbeg;
-					assert(leftQerOffset < MAX_SEQ_LEN_QER * BATCH_SIZE * SEEDS_PER_READ);
+					tmp = s->rbeg - rmax[0];
+					leftRefOffset += tmp;
 					
+					if (numPairsLeft >= *wsize || leftQerOffset >= (*wsize * MAX_SEQ_LEN_QER) || leftRefOffset >= (*wsize * MAX_SEQ_LEN_REF)) {
+						fprintf(stderr, "[LOG][%0.4d] Re-allocating seqBuf (left)\n", tid);
+						*wsize += 1024;
+						uint8_t* seqMem = mmc->seqMem[tid * MAX_LINE_LEN];
+						seqMem = (uint8_t*) realloc(seqMem, *wsize * 

```

Comment:
@arun-sub for this type of reallocs, are you assuming that the alignment offset will remain the same? realloc() can have different start address that can have different offset for alignment. Am I missing something here? Thanks.

Reply:
No I don't assume the alignment is maintained. With realloc() I request (required memory for all seqBuf + 64 bytes) for seqMem. After realloc() returns, I manually align the returned address to the next highest 64-byte aligned address to make sure seqBufLeftRef is aligned to a 64-byte boundary. seqBufLeftQer, seqBufRightRef and seqBufRightQer are using wsize as offsets and are also aligned. I made sure wsize was a multiple of 64 for this.

Reply:
I misunderstood your question. I was trying to avoid the explicit memcpy during reallocation, but my implementation of reallocs is incorrect. This happens when the original malloc for seqMem returns aligned memory but after realloc becomes unaligned. I will try to fix this.

Reply:
It seems hard to avoid the explicit memcpy during reallocation. I just saw that you have implemented mm_realloc in your recent commit, which is a better solution. 

You may have already fixed this in your code, qs and rs need to be reassigned after reallocation ?
https://github.com/bwa-mem2/bwa-mem2/blob/fd12d15c23f781ad84af38a37fbad1b0d8e58ae6/src/bwamem.cpp#L2086

Reply:
Yes, that was incorrect, it should be after the reallocs. Fixed it. Thanks very much for pointing it out.